xdp/src/socket.rs

```diff
@@ -248,6 +242,41 @@ impl<U: Umem> Socket<U> {
     pub fn umem(&mut self) -> &mut U {
         &mut self.umem
     }
+
+    /// Register this AF_XDP socket's fd into the `xsks_map` at the queue id index.
+    /// Safe to call multiple times; later calls overwrite the map entry.
+    pub fn register_in_xskmap(&self, bpf: &mut Ebpf) {
```

Comment:
I don't think we need this method. One can already retrieve the queue using
socket.queue(), what we need to do is implement AsFd for Socket, and then one
can do the registration externally

Reply:
will do!